protocol/msgpack/base_encoder.go

```diff
@@ -187,6 +200,50 @@ func (enc *baseEncoder) encodeFloat64Internal(value float64) {
 	enc.encodeErr = enc.bufEncoder.EncodeFloat64(value)
 }
 
+func (enc *baseEncoder) encodeFloat64SliceInternal(values []float64, encodingType encodingType) {
+	if encodingType == nonPackedEncoding {
+		enc.encodeFloat64SliceNative(values)
+		return
+	}
+	enc.encodeFloat64SlicePacked(values)
+}
+
+// encodeFloat64SliceNative encodes a slice of float64 values using
+// native MessagePack encoding.
+func (enc *baseEncoder) encodeFloat64SliceNative(values []float64) {
+	if enc.encodeErr != nil {
+		return
+	}
+	if enc.encodeErr = enc.bufEncoder.EncodeArrayLen(len(values)); enc.encodeErr != nil {
+		return
+	}
+	for _, v := range values {
+		if enc.encodeErr = enc.bufEncoder.EncodeFloat64(v); enc.encodeErr != nil {
+			return
+		}
+	}
+}
+
+// encodeFloat64SliceNative encodes a slice of float64 values using
+// more compact encoding by encoding the float64 values as byte.
+func (enc *baseEncoder) encodeFloat64SlicePacked(values []float64) {
+	if enc.encodeErr != nil {
+		return
+	}
+	numValues := len(values)
+	numBytes := numValues * numBytesInFloat64
+	if enc.encodeErr = enc.bufEncoder.EncodeBytesLen(numBytes); enc.encodeErr != nil {
+		return
+	}
+	for i := 0; i < numValues; i++ {
+		byteOrder.PutUint64(enc.tmpBuf, math.Float64bits(values[i]))
+		_, enc.encodeErr = enc.bufEncoder.Buffer().Write(enc.tmpBuf)
```

Comment:
Do we want to write the bytes on at a time or would it perhaps be beneficial to write them only into the buffer in the loop and then write the entire buffer once outside the loop?